examples/energy-management-app/energy-management-common/src/ChargingTargetsMemMgr.cpp

```
@@ -0,0 +1,189 @@
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/cluster-objects.h>
+
+#include "ChargingTargetsMemMgr.h"
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::DataModel;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::EnergyEvse;
+
+ChargingTargetsMemMgr::ChargingTargetsMemMgr() : mChargingTargetSchedulesIdx(0), mNumDailyChargingTargets(0)
+{
+    memset(mpListOfDays, 0, sizeof(mpListOfDays));
+}
+
+ChargingTargetsMemMgr::~ChargingTargetsMemMgr()
+{
+    // Free all memory allocated for the charging targets
+    for (uint16_t idx = 0; idx < kEvseTargetsMaxNumberOfDays; idx++)
+    {
+        if (mpListOfDays[idx] != nullptr)
+        {
+            chip::Platform::Delete(mpListOfDays[idx]);
+        }
+    }
+}
+
+void ChargingTargetsMemMgr::PrepareDaySchedule(uint16_t chargingTargetSchedulesIdx)
+{
+    // MUST be called for each entry in DataModel::List<const Structs::ChargingTargetScheduleStruct::Type> chargingTargetSchedules
+    mNumDailyChargingTargets = 0;
+
+    // Should not occur but just to be safe
+    if (chargingTargetSchedulesIdx >= kEvseTargetsMaxNumberOfDays)
+    {
+        ChipLogError(AppServer, "PrepareDaySchedule bad chargingTargetSchedulesIdx %u", chargingTargetSchedulesIdx);
+        return;
+    }
+
+    mChargingTargetSchedulesIdx = chargingTargetSchedulesIdx;
+
+    // Free up any memory associated with this targetSchedule
+    if (mpListOfDays[mChargingTargetSchedulesIdx] != nullptr)
+    {
+        chip::Platform::MemoryFree(mpListOfDays[mChargingTargetSchedulesIdx]);
+        mpListOfDays[mChargingTargetSchedulesIdx] = nullptr;
+    }
+}
+
+void ChargingTargetsMemMgr::AddChargingTarget(const EnergyEvse::Structs::ChargingTargetStruct::Type & chargingTarget)
+{
+    if (mNumDailyChargingTargets < kEvseTargetsMaxTargetsPerDay)
+    {
+        mDailyChargingTargets[mNumDailyChargingTargets++] = chargingTarget;
+    }
```

Log if false. You should not silently fail.